### DISCUSSION TITLE
Weird angular impulse behaviour (rotational inertia)

### DISCUSSION THREAD

**Discussion by highdashtg** [jrouwe/JoltPhysics#855] [Q&A]
Hi, 

I have a simple car (half length = 3, half width = 2, half height = 0.5), with an offset center of mass -half height, copied from the Vehicle Constraints test.  I am trying to rotate it around 3 axes by adding constant angular impulse (AddAngularImpulse).

Maximum angular velocity is set to `6` rad/s and I provide a large enough impulse to test how long it takes for each axis to reach the maximum angular velocity.

Due to (i am guessing here) rotational inertia (roll axis (or the long axis) being less resistant to rotation than pitch axis), it takes lesser amount of time to reach maximum angular velocity on roll axis. This sounds ok and intuitive so far. Note that at this point during my tests, the car does not have any initial rotation applied to it (it looks at the +Z direction).

However, when if i rotate the car around 90 degrees to the left or right (car's forward direction is now +X or -X) and apply a rolling impulse again, now the amount of time it takes to reach maximum angular velocity is not the same. This must be because the roll axis is now not the Z axis, but +X or -X. My guess is this is an issue of AddAngularImpulse requiring you to use "world" space and not the "local" space. This is also true for pitch and yaw.

My intention is to roll (or pitch/yaw) the car around its 3 axes separetely and have them behave the same regardless of the initial rotation in the world. 

Am I thinking this correct?

Thank you

**Comment by jrouwe** [comment]
> However, when if i rotate the car around 90 degrees to the left or right (car's forward direction is now +X or -X) and apply a rolling impulse again, now the amount of time it takes to reach maximum angular velocity is not the same. This must be because the roll axis is now not the Z axis, but +X or -X. My guess is this is an issue of AddAngularImpulse requiring you to use "world" space and not the "local" space. This is also true for pitch and yaw.

* How big is the difference in time between reaching max angular velocity?
* I'm assuming you're comparing the time of roll when the vehicle is not rotated to when the vehicle is rotated 90 degrees? (and not -90 vs 90 degrees)
* I'm also assuming that you adjust AddAngularImpulse to provide an impulse around X instead of Z when the vehicle is rotated?

Any chance you can provide a demo in the Sample framework? (e.g. modify the VehicleConstraintTest demo)

**Comment by highdashtg** [comment]
Hello again,

- Time difference was enough to make a visual difference that would cause players to experience inconsistent behaviour
- I was comparing time from a stand still to a full roll until it reached maximum angular velocity on different world orientations (vehicle forward Z, vehicle forward X etc.)
- Yeah I am rotating my yaw, pitch, roll axes with the vehicle's quaternion.

After not being able to reproduce the issue I was having in Samples demo, I returned back to my application and found that it was an error on my part. I was incorrectly calculating the impulse strength. 

Thank you!
